src/krylov/pcg.c

```diff
@@ -540,11 +562,31 @@ hypre_PCGSolve( void *pcg_vdata,
          break;
       }
       alpha = gamma / sdotp;
-      if (! (alpha > HYPRE_REAL_MIN) )
+      /* alpha should always be greater zero for spd A, spd precond. and nonzero p, r*/
+      /* alpha = <r, Cr> / <p, Ap> */
+      /* for alpha close to 0 x and r will not change much unless x and/or r are close to 0 */
+      if (! (alpha >= HYPRE_REAL_MIN) )
       {
          hypre_error_w_msg(HYPRE_ERROR_CONV, "Subnormal alpha value in PCG");
          if (i == 1) { i_prod = i_prod_0; }
-         break;
+         if (!skip_break)
+         {
+            break;
+         }
+         else if (! (alpha  >= HYPRE_TRUE_MIN))
```

Comment:
I think both the 'alpha >= TRUE_MIN' and 'alpha <= 0.0' tests should always be done here.  That way we have that extra information when --with-print-errors is on, even when skip_break is not set.  Maybe add a boolean variable called `breakout` (or something) that is local to this REAL_MIN check that you set instead of calling `break`, the check its value at the bottom to decide whether or not to break.